Lib/test/test_xmlrpc.py

```diff
@@ -943,8 +943,13 @@ def test_unicode_host(self):
 
     def test_partial_post(self):
         # Check that a partial POST doesn't make the server loop: issue #14001.
-        with contextlib.closing(http.client.HTTPConnection(ADDR, PORT)) as conn:
-            conn.request('POST', '/RPC2 HTTP/1.0\r\nContent-Length: 100\r\n\r\nbye')
+        with contextlib.closing(socket.create_connection((ADDR, PORT))) as conn:
+            conn.send('POST /RPC2 HTTP/1.0\r\n'
+                      'Content-Length: 100\r\n\r\n'
+                      f'bye HTTP/1.1\r\n'
+                      'Host: {ADDR}:{PORT}\r\n'
```

Comment:
It seems like this should be the f-string, not the line before.

Also I don’t get why an existing xmlrpc test is changed by this PR.  It’s testing something else?

Reply:
The previous test was sending a partial payload for which it sent headers as part of the URL that was received by the xmlrpc server using httplib. Now that httplib validates against potential header injection in URL it throws exception and hence this was rewritten to use socket.

Reply:
typo from refactoring. fixed.